doc/sphinx/proof-engine/ltac.rst

```diff
@@ -88,91 +88,72 @@ mode but it can also be used in toplevel definitions as shown below.
-                     : `tactic`
-                     : `qualid` `tacarg` ... `tacarg`
-                     : `atom`
-   atom              : `qualid`
-                     : ()
-                     : `int`
-                     : ( `ltac_expr` )
-   component : `string` | `qualid`
-   message_token     : `string` | `ident` | `int`
-   tacarg            : `qualid`
-                     : ()
-                     : ltac : `atom`
-                     : `term`
-   let_clause        : `ident` [`name` ... `name`] := `ltac_expr`
-   context_rule      : `context_hyp`, ..., `context_hyp` |- `cpattern` => `ltac_expr`
-                     : `cpattern` => `ltac_expr`
-                     : |- `cpattern` => `ltac_expr`
-                     : _ => `ltac_expr`
-   context_hyp       : `name` : `cpattern`
-                     : `name` := `cpattern` [: `cpattern`]
-   match_rule        : `cpattern` => `ltac_expr`
-                     : context [`ident`] [ `cpattern` ] => `ltac_expr`
-                     : _ => `ltac_expr`
-   test              : `int` = `int`
-                     : `int` (< | <= | > | >=) `int`
-   selector          : [`ident`]
-                     : `int`
-                     : (`int` | `int` - `int`), ..., (`int` | `int` - `int`)
-   toplevel_selector : `selector`
-                     : all
-                     : par
-                     : !
-
-.. productionlist:: coq
-   top              : [Local] Ltac `ltac_def` with ... with `ltac_def`
-   ltac_def         : `ident` [`ident` ... `ident`] := `ltac_expr`
-                    : `qualid` [`ident` ... `ident`] ::= `ltac_expr`
+.. insertprodn ltac_expr tactic_atom
+
+.. prodn::
+   ltac_expr ::= @binder_tactic
+   | @ltac_expr4
+   binder_tactic ::= fun {+ @name } => @ltac_expr
+   | let {? rec } @let_clause {* with @let_clause } in @ltac_expr
+   | info @ltac_expr
+   let_clause ::= @name := @ltac_expr
+   | @ident {+ @name } := @ltac_expr
+   ltac_expr4 ::= @ltac_expr3 ; @binder_tactic
+   | @ltac_expr3 ; @ltac_expr3
+   | @ltac_expr3 ; [ {? @multi_goal_tactics } ]
+   | @ltac_expr3 ; [> {? @multi_goal_tactics } ]
+   | @ltac_expr3
+   multi_goal_tactics ::= {? @ltac_expr } %| @multi_goal_tactics
+   | {? @ltac_expr } .. {? %| {*| {? @ltac_expr } } }
+   | @ltac_expr
+   ltac_expr3 ::= try @ltac_expr3
+   | do @int_or_var @ltac_expr3
+   | timeout @int_or_var @ltac_expr3
+   | time {? @string } @ltac_expr3
+   | repeat @ltac_expr3
+   | progress @ltac_expr3
+   | once @ltac_expr3
+   | exactly_once @ltac_expr3
+   | infoH @ltac_expr3
+   | abstract @ltac_expr2
+   | abstract @ltac_expr2 using @ident
+   | only @selector : @ltac_expr3
+   | @ltac_expr2
+   selector ::= {+, @range_selector }
+   | [ @ident ]
+   range_selector ::= @num - @num
+   | @num
+   ltac_expr2 ::= @ltac_expr1 + @binder_tactic
+   | @ltac_expr1 + @ltac_expr2
+   | tryif @ltac_expr then @ltac_expr else @ltac_expr2
+   | @ltac_expr1 || @binder_tactic
+   | @ltac_expr1 || @ltac_expr2
```

Comment:
This could be extracted as a new `ltac_choose` non-terminal.

Reply:
At some point, not too soon, there should be a way to search for grammar, tactics and commands that include the `||` nonterminal.